dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/ChatPromptParser.cs

```diff
@@ -48,18 +52,43 @@ private static bool TryParse(List<PromptNode> nodes, [NotNullWhen(true)] out Cha
     {
         chatHistory = null;
 
-        foreach (var node in nodes)
+        foreach (var node in nodes.Where(IsValidChatMessage))
         {
-            if (IsValidChatMessage(node))
-            {
-                var role = node.Attributes[RoleAttributeName];
-                var content = node.Content!;
+            (chatHistory ??= new()).Add(ParseChatNode(node));
+        }
+
+        return chatHistory is not null;
+    }
 
-                (chatHistory ??= new()).AddMessage(new AuthorRole(role), content);
+    /// <summary>
+    /// Parses a chat node and constructs a ChatMessageContent object.
+    /// </summary>
+    /// <param name="node">The prompt node to parse.</param>
+    /// <returns>A ChatMessageContent object.</returns>
+    private static ChatMessageContent ParseChatNode(PromptNode node)
+    {
+        ChatMessageContentItemCollection items = new();
+        foreach (var childNode in node.ChildNodes)
+        {
+            if (childNode.TagName.Equals(ImageTagName, StringComparison.OrdinalIgnoreCase) && childNode.Content is not null)
```

Comment:
There is no need to check TagName in case if childNode.Content is null, since we can't constuct Content object in this case. We can check if childNode.Content is null at the beginning of the loop and continue if it's true. In this case we will also avoid checking childNode.Content twice (line 73 and line 77).

Reply:
I'll move up content null check. Btw. this is only style optimalization or give real performance benefit?

Reply:
It's style optimization for sure and in terms of performance it depends on which type of chat prompt will be more used - with plain content or Text/Image content.